src/gluonts/dataset/split.py

```diff
@@ -378,20 +376,39 @@ def __iter__(self) -> Generator[Tuple[DataEntry, DataEntry], None, None]:
             max_history=self.max_history,
         )
 
+    def __len__(self):
+        return len(self.dataset) * self.windows
+
     @property
-    def input(self) -> Generator[DataEntry, None, None]:
-        """
-        Iterable over the ``input`` portion of the test data.
-        """
-        for input, _ in self:
-            yield input
+    def input(self) -> "InputDataset":
+        return InputDataset(self)
 
     @property
-    def label(self) -> Generator[DataEntry, None, None]:
-        """
-        Iterable over the ``label`` portion of the test data.
-        """
-        for _, label in self:
+    def label(self) -> "LabelDataset":
+        return LabelDataset(self)
+
+
+@dataclass
+class InputDataset:
+    test_data: TestData
+
+    def __len__(self):
+        return len(self.test_data)
+
+    def __iter__(self):
+        for input_, label in self.test_data:
+            yield input_
```

Comment:
I think it's fine to overwrite `input`. Also, I would use `_` as a prefix to indicate that the value is not used.

```suggestion
        for input, _label in self.test_data:
            yield input
```